modules/carto/src/layers/spatial-index-tile-layer.js

```diff
@@ -0,0 +1,47 @@
+import {TileLayer} from '@deck.gl/geo-layers';
```

Comment:
Write new classes in TypeScript?

Reply:
As the other files were in JS in this directory I stuck with that, but the plan is to migrate this next week

Reply:
Our plan is to do it right after this PR. I know in the future we should develop directly in typescript, but we were in hurry for this to do QA in other pieces of our platform.

Added to the TODO of spatial indexes: https://github.com/visgl/deck.gl/issues/6912#issuecomment-1129682952 to make sure we don't forget about it